### PR TITLE
Pr 3209

### PR DESCRIPTION
**Pull request by ndellingwood**
To supersede PR #3209 - updates to the superlu_dist 5.4.0 changes submitted by @balay to guard for older versions of superlu_dist.

### CODE REVIEW

File: packages/amesos/src/Amesos_Superludist.cpp [hunk 1]
```diff
@@ -135,7 +135,11 @@ Amesos_Superludist::Amesos_Superludist(const Epetra_LinearProblem &prob) :
   Equil_ = true;
   ColPerm_ = "MMD_AT_PLUS_A";
   perm_c_ = 0;
+#if (SUPERLU_DIST_MAJOR_VERSION > 5) || ( SUPERLU_DIST_MAJOR_VERSION == 5 && SUPERLU_DIST_MINOR_VERSION == 4)
```
Comment (balay) [inline packages/amesos/src/Amesos_Superludist.cpp]: This change is fine [as there is no 5.5 release] - but I  think the code might be cleaner with:
```
#if (SUPERLU_DIST_MAJOR_VERSION > 5) || ( SUPERLU_DIST_MAJOR_VERSION == 5 && SUPERLU_DIST_MINOR_VERSION > 3)
```

**Comment by balay** [comment]
BTW: @ndellingwood ,   I amended your commit [with my suggested changes] and pushed to  #3209 - if you'd like to pick that up..

File: packages/amesos/src/Amesos_Superludist.cpp (hunk 1 above)
Comment (ndellingwood) [reply packages/amesos/src/Amesos_Superludist.cpp]: Made change in recent push.

**Comment by ndellingwood** [comment]
@balay thanks for review comments, I made the changes you suggested to the PR and fixed the Amesos2 check `SUPERLU_DIST_MAJOR_VERSION` to match your previous fix you pointed out which fixed the merge conflict. Merged with develop and pushed changes. If you'd prefer your PR #3209 be tested/merged instead feel free to reopen and close this one.

**Comment by balay** [comment]
> @balay thanks for review comments, I made the changes you suggested to the PR and fixed the Amesos2 check `SUPERLU_DIST_MAJOR_VERSION` to match your previous fix you pointed out which fixed the merge conflict. Merged with develop and pushed changes. If you'd prefer your PR #3209 be tested/merged instead feel free to reopen and close this one.

Either PR is fine.

For this one - you might also want to fix packages/amesos/test/TestOptions/TestSuperludist.cpp wtith SUPERLU_DIST_MINOR_VERSION changes

**Comment by balay** [comment]
Note: these fixes [whichever PR gets merged] would have to eventually go into branch `trilinos-release-12-14-branch`
cc: @jwillenbring

**Comment by ndellingwood** [comment]
@balay 

> For this one - you might also want to fix packages/amesos/test/TestOptions/TestSuperludist.cpp wtith SUPERLU_DIST_MINOR_VERSION changes

Yep, I missed that file. We'll close this one and use PR #3209, it has the cleaner commit history and I think it hit the autotester first so we can merge it in faster.

**Pull request closed without merging**